agent/fw_laravel_queue.c

```diff
@@ -624,8 +579,52 @@ NR_PHP_WRAPPER(nr_laravel_queue_workcommand_fire) {
    * aren't executed if we're not actually in a transaction.
    *
    * So instead, what we'll do is to keep recording, but ensure that we ignore
-   * the transaction after WorkCommand::fire() has finished executing, at which
-   * point no more jobs can be run.
+   * the transaction after WorkCommand::handle() has finished executing, at
+   * which point no more jobs can be run.
+   */
+
+  /*
+   * Start listening for jobs.
+   */
+  nr_php_wrap_user_function_before_after_clean(
+      NR_PSTR("Illuminate\\Queue\\Worker::process"),
+      nr_laravel_queue_worker_process_before,
+      nr_laravel_queue_worker_process_after, NULL);
+
+  NR_PHP_WRAPPER_CALL;
+}
+NR_PHP_WRAPPER_END
+
+NR_PHP_WRAPPER(nr_laravel_queue_workcommand_handle_after) {
+  NR_UNUSED_SPECIALFN;
+  (void)wraprec;
+
+  NR_PHP_WRAPPER_REQUIRE_FRAMEWORK(NR_FW_LARAVEL);
+  /*
+   * Stop recording the transaction and throw it away.
+   */
+  nr_php_txn_end(1, 0 TSRMLS_CC);
```

Comment:
Do we not need to then start a new transaction here?

Reply:
No, we want to record individual transactions for each job that is executed, but don't want to record a transaction for the actual queue:work command, since it spends most of its time sleeping.  Calling `nr_php_txn_end` here ensures that we ignore the transaction after WorkCommand::handle() has finished executing, at which point no more jobs can be run.  All of the job and txn naming is handled in the `Worker::process` wrapper which has already completed by the time it reaches this point.

Reply:
WorkCommand::handle was removed entirely for PHP 8+. (but tests before removing showed that it did indeed need to restart the txn)